docs/spec/v1beta3/providers.md

```diff
@@ -1187,8 +1187,35 @@ stringData:
 ### TLS certificates
 
 `.spec.certSecretRef` is an optional field to specify a name reference to a
-Secret in the same namespace as the Provider, containing the TLS CA certificate.
-The secret must be of type `kubernetes.io/tls` or `Opaque`.
+Secret in the same namespace as the Provider, containing TLS certificates for
+secure communication. The secret must be of type `kubernetes.io/tls` or `Opaque`.
+
+#### Supported configurations
+
+- **CA-only**: Server authentication (provide `ca.crt` only)
+- **mTLS**: Client certificate authentication (provide `tls.crt` + `tls.key`, optionally with `ca.crt`)
+
+#### Providers supporting client certificate authentication
+
+The following webhook-based providers support client certificate authentication:
+
+| Provider Type        | Description                    |
+|---------------------|--------------------------------|
+| `alertmanager`      | Prometheus Alertmanager        |
+| `discord`           | Discord webhooks               |
+| `forwarder`         | Generic forwarder              |
+| `grafana`           | Grafana annotations API        |
+| `matrix`            | Matrix rooms                   |
+| `msteams`           | Microsoft Teams                |
+| `opsgenie`          | Opsgenie alerts                |
+| `pagerduty`         | PagerDuty events               |
+| `rocket`            | Rocket.Chat                    |
+| `slack`             | Slack API                      |
+| `webex`             | Webex messages                 |
+
+Support for client certificate authentication is being expanded to additional providers over time.
+
+**Note**: Git platform providers (GitHub, GitLab, etc.) use third-party SDKs that handle TLS configuration differently.
```

Comment:
```suggestion
```

I don't think we need this here, feels like an implentation detail that doesn't matter much. Unless, for example, if there's a known exception among the SDKs that makes the behavior notoriously different from the others. But that will matter only after we go through the work of adding the functionality there, then we'll know what exceptions we need to document if any.

Reply:
Fixed via [b2689c4](https://github.com/fluxcd/notification-controller/pull/1137/commits/b2689c46d389e1dc383b3aaa55e9d45b883cbeea) .